packages/xo-web/src/xo-app/vm/tab-advanced.js

```diff
@@ -641,17 +619,43 @@ export default connectStore(() => {
                 <th>{_('resourceSet')}</th>
                 <td>
                   {isAdmin ? (
-                    <SelectResourceSet
-                      onChange={resourceSet =>
-                        editVm(vm, {
-                          resourceSet:
-                            resourceSet != null ? resourceSet.id : resourceSet,
-                        })
-                      }
-                      value={vm.resourceSet}
-                    />
+                    <div className='input-group'>
+                      <SelectResourceSet
+                        onChange={resourceSet =>
+                          editVm(vm, {
+                            resourceSet:
+                              resourceSet != null ? resourceSet.id : resourceSet,
+                          })
+                        }
+                        value={vm.resourceSet}
+                      />
+                      {vm.resourceSet !== undefined && (
+                        <span className='input-group-btn'>
+                          <ActionButton
+                            btnStyle='primary'
+                            handler={shareVmProxy}
+                            handlerParam={vm}
+                            icon='vm-share'
+                            style={SHARE_BUTTON_STYLE}
+                            tooltip={_('vmShareButton')}
+                          />
+                        </span>
+                      )}
+                    </div>
                   ) : vm.resourceSet !== undefined ? (
-                    <ResourceSetItem id={vm.resourceSet} />
+                    <span>
+                      <ResourceSetItem id={vm.resourceSet} />{' '}
+                      {canAdministrate && (
```

Comment:
Why do you need this if we assumed that Self Service and ACLs shouldn't be used together?